docs(instances): add reboot from faulty kernel (#1034)
scaleway/docs-content: compute/instances/troubleshooting/reboot-from-faulty-kernel.mdx

---
meta:
  title: Rebooting from a faulty kernel
  description: This page helps you troubleshoot issues related to the installation of a faulty kernel
content:
  h1: Rebooting from a faulty kernel
  paragraph: This page helps you troubleshoot issues related to the installation of a faulty kernel
tags: kernel faulty-kernel reboot bugfix
dates:
  validation: 2022-11-23
  posted: 2022-11-23
categories:
  - compute
---

> In rare cases your system does fails to boot or boots incorrectly after the installation of a new kernel version, you can use [rescue mode](/compute/instances/concepts/#rescue-mode) to boot into the previous working kernel.

Reviewer: [Rescue mode](/compute/instances/concepts/#rescue-mode) allows you to boot your Instance into a temporary operating system, often the previous version of the working kernel. Rescue mode is often used to diagnose and resolve issues such as networking issues, firewall configurations...etc.

[    4.224435] cloud-init[542]:     retval = util.log_time(
[    4.225251] cloud-init[542]:   File "/usr/lib/python3/dist-packages/cloudinit/util.py", line 2348, in log_time
[    4.226838] cloud-init[542]:     ret = func(*args, **kwargs)
[    4.227743] cloud-init[542]:   File "/usr/lib/python3/dist-packages/cloudinit/cmd/main.py", line 600, in status_wrapper
[    4.229478] cloud-init[542]:     util.ensure_dirs((data_d, link_d,))
[    4.230501] cloud-init[542]:   File "/usr/lib/python3/dist-packages/cloudinit/util.py", line 1534, in ensure_dirs
[    4.232014] cloud-init[542]:     ensure_dir(d, mode)
[    4.232540] cloud-init[542]:   File "/usr/lib/python3/dist-packages/cloudinit/util.py", line 1591, in ensure_dir
[    4.233980] cloud-init[542]:     chmod(path, mode)
[    4.234756] cloud-init[542]:   File "/usr/lib/python3/dist-packages/cloudinit/util.py", line 1830, in chmod
[    4.235672] cloud-init[542]:     os.chmod(path, real_mode)
[    4.236227] cloud-init[542]: OSError: [Errno 30] Read-only file system: '/var/lib/cloud/data'
[FAILED] Failed to start Apply the …ngs specified in cloud-config.

> ```

Reviewer: Je trouve ca bizarre de mettre cet exemple dans l'intro, est-ce qu'on peut pas rajouter une section "Examples" ? ou quelque chose comme ca ?

    8 15 126976 sda15
    # mount /dev/sda1 /mnt
    # mount /dev/sda15 /mnt/boot/efi
    # mount -o bind /sys /mnt/sys
    # mount -o bind /proc /mnt/proc
    # mount -o bind /dev /mnt/dev
    ```
3. Once mounted, use the `chroot` command to get into your Instances' root filesystem. You can then change the `GRUB_DEFAULT` value to boot using the previous kernel:
    ```sh
    chroot /mnt
    nano /etc/default/grub
    ```
    <Message type="tip">
        In the example above we use `nano` as text editor. Feel free to use your favorite text editor to edit the file. 
    </Message>
    Change the value of `GRUB_DEFAULT` to `"1 > 2":
    ```sh
    # head /etc/default/grub
    # If you change this file, run 'update-grub' afterwards to update
    # /boot/grub/grub.cfg.
    # For full documentation of the options in this file, see:
    #   info -f grub -n 'Simple configuration'
    
    GRUB_DEFAULT="1>2"
    GRUB_TIMEOUT=5
    GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian`
    GRUB_CMDLINE_LINUX_DEFAULT=""
    GRUB_CMDLINE_LINUX="console=tty0 console=ttyS0,115200 earlyprintk=ttyS0,115200 consoleblank=0"
    ```
    Save the file and exit the text editor. 

> 4. Update the `grub` boatloader, so the change of the kernel is taken into account:

Reviewer: ```suggestion
4. Update the `grub` boatloader` to apply the kernel's update
```

[FAILED] Failed to start Apply the …ngs specified in cloud-config.
```

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [created an Instance](/compute/instances/how-to/create-an-instance)
  - You have installed and configured the Scaleway [CLI-Tools](/compute/instances/api-cli/creating-managing-instances-with-cliv2) on your local computer
</Message>

1. Switch the Instance's `boot-type` to `rescue` and reboot your Instance into rescue mode using the CLI-Tools: 
    ```
    # scw instance server update {Instance_ID} boot-type=rescue
    # scw instance server reboot {Instance_ID}
    ```
    <Message type="note">
        Replace `{Instance_ID}` with the unique ID of your Instance, e.g. `0500ebd2-d70d-49af-a969-3ac09b6f7fff`.
    </Message>
2. Once the Instance is rebooted, log into your Instance using [SSH](/compute/instances/how-to/connect-to-instance/) and setup the environment to be able to chroot into it:
    ```sh
    # cat /proc/partitions
    major minor #blocks name

    8 0 9765625 sda
    8 1 9634536 sda1
    8 14 3072 sda14
    8 15 126976 sda15
    # mount /dev/sda1 /mnt

>     mount: /mnt: WARNING: source write-protected, mounted read-only.

Reviewer 2: This line should be removed as this is an error from the capture

>     mount: /mnt: WARNING: source write-protected, mounted read-only.

Author: ```suggestion
```
